app/apphandlers/processorAppHandler.go

```diff
@@ -223,6 +222,23 @@ func (a *processorApp) StartRudderCore(ctx context.Context, options *app.Options
 
 	adaptiveLimit := payload.SetupAdaptiveLimiter(ctx, g)
 
+	var geoEnricher enricher.PipelineEnricher
+	if config.GetBool("GeoEnrichment.enabled", false) {
+
+		dbProvider, err := enricher.NewGeoDBProvider(config.Default, a.log)
+		if err != nil {
+			return fmt.Errorf("creating new instance of db provider: %w", err)
```

Comment:
[minor] May be we can we more specific here?
```suggestion
			return fmt.Errorf("creating new instance of geo db provider: %w", err)
```